Runtime/Scripts/DataCollection/RGDataCollection.cs

```diff
@@ -298,14 +341,22 @@ await RGServiceManager.GetInstance()
             {
                 Cleanup(clientId);
                 // If there are no more bots, cleanup everything else
-                if (_clientIdToBots.Count == 0)
+                if (_clientIdToState.IsEmpty)
```

Comment:
IsEmpty is _slightly_ optimized in the case where the dictionary is not empty. `Count` has to acquire all the locks in the dictionary, whereas `IsEmpty` can do a fast lock-free check to see if there are any elements and return immediately if there are.